docs/experimental/extendable-param-files.md

```diff
@@ -0,0 +1,63 @@
+# Using the Extendable Bicep Parameters Feature (Experimental!)
+
+## What is it?
+
+Extendable Bicep Parameter Files is a feature that allows you to extend `.bicepparam` files from another `.bicepparam` file in order to reuse parameters across multiple deployments.
+
+When using extendable bicep parameter files, you will have a `base.bicepparam` file that can be used by multiple `.bicep` and `.bicepparam` files.
+
+The `base.bicepparam` file will link to a `shared.bicepparam` file, which will refer to a `main.bicep` file and your `base.bicepparam file`. In this case, you will be able to reuse and refer to this `base.bicepparam` file multiple times.
+
+## Using
+
+`main.bicep` This is your main bicep file, which will define your parameters for deployment.
+
+```bicep
+param foo string
+param foo string
+```
+
+`base.bicepparam` This is your base bicepparam file, which can be reused by multiple shared.bicepparam files and in multiple deployments.
+
+```bicep
+using none
+
+param foo = 'foo'
+param bar = 'bar'
+```
+
+`shared.bicepparam` This is your shared bicepparam file, which will refer to one main.bicep file and one base.bicepparam file. Any parameter definition in this file will override all previous definitions.
+
+```bicep
+using 'main.bicep'
+
+extends 'base.bicepparam'
+
+param foo = 'bar'
+param baz = foo
+```
+
+Compiled json output
```

Comment:
I understand your intention here to keep this output simple, but if we are going to explicitly say this is the compile JSON output, then it should be valid JSON. The below sample is valid Bicep, but not valid JSON.

Reply:
I wonder if we should even talk about JSON at all? I think some customers may not be used to thinking in terms of "compilation" & "JSON".

We could potentially just say e.g.:

Resolved values:

| Param | Value |
| -- | -- |
| foo | `'bar'` |
| bar | `'bar'` |
| baz | `'bar'` |

Reply:
I agree with Anthony here.

```json
"parameters": {
  "namePrefix": {
    "type": "string"
    "defaultValue": "extend"
  },
  "location": {
    "type": "string"
    "defaultValue": "westus"
  }
  "symbolicName": {
    "type": "string"
    "defaultValue": "test"
  }
}
``` 
this seems a little too complicated to read so I'll go with a table format.